### ISSUE TITLE
Automatically handle parallel queries with an async `Lock`

### ISSUE THREAD

**Issue by LiraNuna** [aio-libs/aiomysql#852]
### Is your feature request related to a problem?

When performing two "parallel" queries via `asyncio.gather`, an error is thrown: 
```RuntimeError: readexactly() called while another coroutine is already waiting for incoming data```

### Describe the solution you'd like

I have implemented a simple yet effective way to remedy this issue however it's quite "gross", requiring monkey patching as I am unable to perform the required changes in a type safe manner:

```python
class LockedCursor(Cursor):
    def get_lock(self) -> Lock:
        if not hasattr(self.connection, 'lock'):
            self.connection.lock = Lock()

        return self.connection.lock

    async def execute(self, query_string: str, args: Optional[Iterable[Any]] = None):
        async with self.get_lock():
            return await super().execute(query_string, args)
```

I recommend this feature to be standard on the base cursor, as it's often time impossible to avoid parallel queries happening at the same time.

### Describe alternatives you've considered

Another option would involve the connection holding an instance of `asyncio.Queue` to orchestrate any parallel queries going out, however I personally believe this is overkill and is not as simple to understand as a lock. 

Another option I considered is subclassing each of the `Pool`, `Connection` and `Cursor` class to inject `LockedCursor` when calling `connection.cursor`, however that has proven to be very difficult and full of boilerplate and repetitive code to override the constructor and other methods simply to add a lock.

### Code of Conduct

- [X] I agree to follow the aio-libs Code of Conduct

**Comment by LiraNuna** [comment]
After a few days with this solution I don't think it's not sufficient to JUST lock `execute`, I think every operation should be locked, such as `fetchall`, `fetchone` etc.

**Comment by Nothing4You** [comment]
Hi,
the simplest solution for this that would require no code changes is to just [`acquire()`](https://aiomysql.readthedocs.io/en/latest/pool.html#Pool.acquire) a connection from the pool for each task.
By providing appropriate `minsize` and `maxsize` values the pool will automatically take care of creating the necessary amount of connections and it will not return on `acquire()` until a connection is available.

**Comment by LiraNuna** [comment]
I don't think this is solution will work if need to work within the same connection transaction, which is my requirement. 
On our server, we use a connection per request where eat connection is also a transaction we can rollback if the request fails (due to an exception or other error).

**Comment by Nothing4You** [comment]
Indeed, transactions will get lost with that.

**Comment by LiraNuna** [comment]
Any progress on this? An `asyncio.Lock` is an elegant solution that is easy to implement. I'm willing to contribute a PR if necessary, I just want to make sure it will get quickly reviewed if I spend the time on this.
